addons/l10n_de/models/account_account.py

```diff
@@ -6,7 +6,11 @@ class AccountAccount(models.Model):
     _inherit = ['account.account']
 
     def write(self, vals):
-        if 'code' in vals and 'DE' in self.company_id.account_fiscal_country_id.mapped('code'):
+        if (
+            'code' in vals
+            and 'DE' in self.company_id.account_fiscal_country_id.mapped('code')
+            and any(a.code != vals['code'] for a in self)
```

Comment:
by the way, I would have preferred saying acc or account instead of a :D but whatever, didn't complain ^^